zerver/lib/event_schema.py

```diff
@@ -1516,13 +1516,15 @@ def check_update_global_notifications(
     assert isinstance(setting, setting_type)
 
 
+# user_id field is null for embedded variant of update_message
 update_message_required_fields = [
     ("type", Equals("update_message")),
-    ("user_id", int),
+    ("user_id", OptionalType(int)),
     ("edit_timestamp", int),
     ("message_id", int),
     ("flags", ListType(str)),
     ("message_ids", ListType(int)),
+    ("rendering_only", bool),
```

Comment:
And we'll want to check that this is true or false in the right situations. In the context of the existing `check_update_message` design, the simplest way to do that is probably to check that it equals your new `is_embedded_update_only` argument.